### PR TITLE
minesweeper-api-change

### LINKED ISSUE #2046: Minesweeper: deviance from problem specs and other tweaks
I noticed the Minesweeper exercise deviates from the problem specs in a way that is not documented. The [original exercise](https://github.com/exercism/problem-specifications/blob/main/exercises/minesweeper/description.md) asks the student the student to fill each square of a board with the number of mines adjacent to that square. The Go version of the exercise deviates from this spec by adding a border to the board. However, there's nothing in the description talking about this additional border. The tests check if the border is "correct", without explicit saying what makes a border correct (the student has to look at the tests and do some guesswork).

Overall, this exercise feels like it could use "more love". What I think we should look at:

- [ ] Decide if we want to keep the deviance from problem specs or not. If we want to keep it, we must include an additional description  mentioning the board border and we should add more custom tests checking if the border is correct (as of right now, they allow some incorrect borders). I think we should adhere to the problem specs, as I don't think adding a border to the board adds much to the exercise, but I'm interested to see other opinions.
- [ ] Improve [the tests](https://github.com/exercism/go/blob/main/exercises/practice/minesweeper/minesweeper_test.go) organization. Might be worth to organize the tests into named sub-tests that can be picked up by the test runner, like what happens with other exercises. This would improve the experience for someone using the online editor and the testing experience in general.
- [ ] [The board](https://github.com/exercism/go/blob/fddd098d153121ae0d302ec1ae91f73655e57a3c/exercises/practice/minesweeper/board.go#L5) is a `[][]byte`, which I find a bit weird, not sure if anyone else feels the same. Seems like a more natural choice would be `[][]rune`. For some solutions this makes no difference, but there are other solutions where students make smaller functions, and I find a bit weird seeing them receiving `byte`s representing characters of the board. I feel like working with `rune`s would be more appropriate.

### PR DESCRIPTION
**Pull request by norbs57**
Resolves #2046
Minesweeper exercise

- Changed API to `func Annotate(board []string) []string` 
- Replaced tests with canonical tests
- Defined `testAnnotate` using `T.run`
- Replaced sample solution with two new examples 
- Updated `config.json`

### CODE REVIEW

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
Thanks for working on this!

Left some suggestions.

File: exercises/practice/minesweeper/minesweeper_test.go [hunk 1]
```diff
@@ -1,193 +1,170 @@
...
+			got := Annotate(tc.minefield)
+			want := tc.expected
+			if !reflect.DeepEqual(want, got) {
+				t.Fatalf("expected: %v, got: %v", want, got)
+				t.FailNow()
+			}
+		})
 	}
-}
+}
```
[the remark this answers is not in the document]
Comment (norbs57) [reply exercises/practice/minesweeper/minesweeper_test.go:170]: Will do. It seems a bit of a chore to keep checking this, so I wonder if there is an automatic way to add a newline at the end if absent?

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
I like the shape this is taking!

Some final suggestions that escaped my initial review.

File: exercises/practice/minesweeper/minesweeper_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrerfcsantos) [reply exercises/practice/minesweeper/minesweeper_test.go]: I agree we should see a good way to automate tasks like this. Right now, there isn't.

**Review by andrerfcsantos — APPROVED** [review]
Thanks for this!

Merged 🚀

**Pull request merged**